wasm/src/strings.c

```diff
@@ -349,6 +349,14 @@ opa_value *opa_strings_split(opa_value *a, opa_value *b)
     }
 
     opa_string_t *s = opa_cast_string(a);
+
+    if (s->len == 0)
```

Comment:
This won't fix the more general case where len(s) < len(d), i.e., s = "x" and d = "xx". You could change the final for loop to be: `for (int i = 0; s->len >= d->len && i <= (s->len - d->len); )` to avoid the overflow causing the issue.

Reply:
Nice catch. Fixed.